pkg/backend/aws/ecs.go

```diff
@@ -358,23 +359,33 @@ func (ab *Backend) getTaskId(taskArn string) (string, error) {
 	return arnSegments[len(arnSegments)-1], nil
 }
 
-func (ab *Backend) getLogGroupAndStreamName(taskDefinition ecstypes.TaskDefinition, taskId string, containerName string) (string, string, error) {
+func (ab *Backend) getLogGroupAndStreamName(taskDefinition ecstypes.TaskDefinition, task ecstypes.Task, taskId string, containerName string) (string, string, error) {
 	logGroup := ""
 	logStreamName := ""
+	containerMap := map[string]ecstypes.Container{}
+	for _, container := range task.Containers {
+		containerMap[*container.Name] = container
+	}
 	for _, containerDefinition := range taskDefinition.ContainerDefinitions {
 		// If container name is specified, we only look at that container
 		if len(containerName) > 0 && (*containerDefinition.Name != containerName) {
 			continue
 		}
+		container, ok := containerMap[containerName]
+		if !ok {
+			continue
+		}
 		logGroup, ok := containerDefinition.LogConfiguration.Options[AwsLogsGroup]
 		if !ok {
 			continue
 		}
-		logStreamName := taskId
+		logStreamName := *container.RuntimeId
+
 		logPrefix, ok := containerDefinition.LogConfiguration.Options[AwsLogsStreamPrefix]
 		if ok {
 			logStreamName = path.Join(logPrefix, *containerDefinition.Name, taskId)
```

Comment:
why do we still use `taskId` here?

Reply:
EC2 profile tasks rely on taskId, we can completely deprecate it